### ISSUE TITLE
Bug using masks in v4.x in old devices

### ISSUE THREAD

**Issue by daniel-dia** [pixijs/pixi.js#3667]
Hello. In version PIXI 4.x, 

When a Mask is used, all screen becomes messy. Here is a video comparing two devices.
Both using cocoon js. 
The left (iPhone) is running using multi texture batching.
The right one (android) is an old device running with this configurations

```
PIXI["settings"].SPRITE_MAX_TEXTURES = 1;
PIXI["settings"].PRECISION = PIXI.PRECISION.MEDIUM;
PIXI.glCore.VertexArrayObject.FORCE_NATIVE = true;
```

 (the mask is in the game header)
https://www.youtube.com/watch?v=ffJtwwnZFXE

When the header is not visible anymore at game end, all sprites are correctly rendered.
When the header is visible again at game restart, some sprites are partially drew.

The code I used to create a mask is:
```
var levelBar = gameui.AssetsManager.getBitmap("bonus_bar");
levelBar.mask = new PIXI.Graphics().beginFill(0xFF0000).drawRect(0, 0, 939, 57).endFill();
this.levelBar = levelBar;
this.addChild(levelBar);
this.addChild(levelBar.mask);
```

this error does not happens in PIXI v3

What can I do to solve it.

**Comment by Dimous** [comment]
Try without PIXI.glCore.VertexArrayObject.FORCE_NATIVE = true;

**Comment by artsjedi** [comment]
Unfortunately using `PIXI.glCore.VertexArrayObject.FORCE_NATIVE = true` or not does not affect this bug.

**Comment by Dimous** [comment]
Out of curiosity added this option to my project and mask stopped working, so I decided it would cause this bug.
_Win x64 10, FF 51.0.1_

**Comment by artsjedi** [comment]
After a bunch o tests i figure out that the mask only works if the masked object is the top most object. It means, there is no one object in front of the masked one. os else the display is messy.

Maybe is some bug in 'turning off' some clamp parameter after the mask is applied.

**Comment by GoodBoyDigital** [comment]
hey @artsjedi do you still get the issue if you change the mask for to a circle (just to test a theory!) 
Also an isolated example would make it super easy for us to debug and fix for you 👍 

Cheers!

**Comment by artsjedi** [comment]
Hello @GoodBoyDigital , I changed from rectangle to circle and then to polygon. It does not affects the issue.

As you requested, I'm uploaded a isolated exemple of this issue in this zip file. 
Just open install cocoon in a old android phone device and then opens this file. 

[CocoonPackage.zip](https://github.com/pixijs/pixi.js/files/737990/CocoonPackage.zip)
 
here is the my screen captured showing the issue.
https://youtu.be/Ua_UVoieho8

This exemple changes the "power bar" mask and sometimes adds a pause menu in front of it.
Note that the issue only appears when the "pause menu" overlays the "power bar".

This project I used typescript to compile all .ts into a single script.js file. All ts source files are into the src folder.

If you have any questions, do not hesitate to contact me.
Daniel
